src/EPR.SubsidiaryBulkUpload.Application/Services/BulkSubsidiaryProcessor.cs

```diff
@@ -1,24 +1,36 @@
 ﻿using EPR.SubsidiaryBulkUpload.Application.DTOs;
+using EPR.SubsidiaryBulkUpload.Application.Extensions;
 using EPR.SubsidiaryBulkUpload.Application.Models;
+using EPR.SubsidiaryBulkUpload.Application.Services.Interfaces;
 using Microsoft.Extensions.Logging;
 
 namespace EPR.SubsidiaryBulkUpload.Application.Services;
 
-public class BulkSubsidiaryProcessor(ISubsidiaryService organisationService, ICompaniesHouseDataProvider companiesHouseDataProvider, ILogger<BulkSubsidiaryProcessor> logger)
+public class BulkSubsidiaryProcessor(ISubsidiaryService organisationService, ICompaniesHouseDataProvider companiesHouseDataProvider, ILogger<BulkSubsidiaryProcessor> logger, INotificationService notificationService)
     : IBulkSubsidiaryProcessor
 {
+    private const string SubsidiaryBulkUploadMismatchedProgress = "Subsidiary bulk upload progress";
+    private const string SubsidiaryBulkUploadMismatchedErrors = "Subsidiary bulk upload errors. Mismatched subsidiaries.";
     private readonly ILogger<BulkSubsidiaryProcessor> _logger = logger;
     private readonly ISubsidiaryService organisationService = organisationService;
     private readonly ICompaniesHouseDataProvider companiesHouseDataProvider = companiesHouseDataProvider;
+    private readonly INotificationService _notificationService = notificationService;
 
-    public async Task Process(IEnumerable<CompaniesHouseCompany> subsidiaries, CompaniesHouseCompany parent, OrganisationResponseModel parentOrg, Guid userId)
+    public async Task Process(IEnumerable<CompaniesHouseCompany> subsidiaries, CompaniesHouseCompany parent, OrganisationResponseModel parentOrg, UserRequestModel userRequestModel)
     {
+        // these are the subsidiaries which exists in RPD
```

Comment:
Remove the comments?

Reply:
there are lots of comments between line 20 and 82. its quite complex code. these comments would be useful for anyone reading the code or taking over this application.